### PR TITLE
Correct metadata name/unit/dimension 

### PR DESCRIPTION
**Pull request by mzhangw**
This PR includes metadata name/unit/dimension correctness. It does not change answers.
The associated PR:https://github.com/NCAR/ccpp-physics/pull/496

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>

## Testing

How were these changes tested?  
What compilers / HPCs was it tested with?  
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/nems/pull/<pr_number>
- waiting on noaa-emc/fv3atm/pull/<pr_number>

### CODE REVIEW

**Comment by climbfuji** [comment]
Note: this PR will be pulled into a feature branch feature/transition-to-capgen-1 in the NCAR GitHub repo, which collects several non-answer-changing PRs for that effort of transitioning to `cap_gen.py`.

File: gfsphysics/GFS_layer/GFS_typedefs.F90 [hunk 1]
```diff
@@ -1055,6 +1055,7 @@ real (kind=kind_phys), pointer :: prsik (:,:) => null()   !< Exner function at i
 !--- tracer handling
     character(len=32), pointer :: tracer_names(:) !< array of initialized tracers from dynamic core
     integer              :: ntrac           !< number of tracers
+    character(len=20), pointer :: fscav_aero(:)   !< aerosol scavenging factors
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.F90]: As far as I can see, all changes to `GFS_typedefs.F90` need to be reverted.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 2]
```diff
@@ -2469,14 +2469,14 @@
   dimensions = ()
   type = integer
 [iovr_sw]
-  standard_name = flag_for_max_random_overlap_clouds_for_shortwave_radiation
-  long_name = sw: max-random overlap clouds
+  standard_name = flag_for_cloud_overlapping_method_for_shortwave_radiation
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 3]
```diff
@@ -2469,14 +2469,14 @@
...
+  long_name = control flag for cloud overlapping method for SW
   units = flag
   dimensions = ()
   type = integer
 [iovr_lw]
-  standard_name = flag_for_max_random_overlap_clouds_for_longwave_radiation
-  long_name = lw: max-random overlap clouds
+  standard_name = flag_for_cloud_overlapping_method_for_longwave_radiation
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 4]
```diff
@@ -2487,14 +2487,14 @@
   dimensions = ()
   type = integer
 [isubc_sw]
-  standard_name = flag_for_sw_clouds_without_sub_grid_approximation
-  long_name = flag for sw clouds without sub-grid approximation
+  standard_name = flag_for_subcolumn_cloud_approximation_for_shortwave_radiation
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 5]
```diff
@@ -2487,14 +2487,14 @@
...
+  long_name = flag for subcolumn cloud approximation for shortwave radiation
   units = flag
   dimensions = ()
   type = integer
 [isubc_lw]
-  standard_name = flag_for_lw_clouds_without_sub_grid_approximation
-  long_name = flag for lw clouds without sub-grid approximation
+  standard_name = flag_for_subcolumn_cloud_approximation_for_longwave_radiation
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 6]
```diff
@@ -4035,7 +4035,7 @@
   units = flag
   dimensions = (number_of_chemical_tracers)
   type = logical
-[fscav]
+[fscav_sas]
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 7]
```diff
@@ -4572,14 +4572,14 @@
   units = none
   dimensions = (horizontal_dimension)
   type = integer
-  active = (flag_for_lw_clouds_without_sub_grid_approximation == 2 .or. flag_for_sw_clouds_without_sub_grid_approximation == 2)
+  active = (flag_for_subcolumn_cloud_approximation_for_longwave_radiation == 2 .or. flag_for_subcolumn_cloud_approximation_for_shortwave_radiation == 2)
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 8]
```diff
@@ -4572,14 +4572,14 @@
...
 [icsdlw]
   standard_name = seed_random_numbers_lw
   long_name = random seeds for sub-column cloud generators lw
   units = none
   dimensions = (horizontal_dimension)
   type = integer
-  active = (flag_for_lw_clouds_without_sub_grid_approximation == 2 .or. flag_for_sw_clouds_without_sub_grid_approximation == 2)
+  active = (flag_for_subcolumn_cloud_approximation_for_longwave_radiation == 2 .or. flag_for_subcolumn_cloud_approximation_for_shortwave_radiation == 2)
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 9]
```diff
@@ -4602,23 +4602,23 @@
   type = real
   kind = kind_phys
 [in_nm]
-  standard_name = in_number_concentration
-  long_name = IN number concentration
-  units = kg-1?
+  standard_name = ice_nucleation_number
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta:4605]: Same as for ccpp-physics, has this change been coordinated with @grantfirl and @ligiabernardet?

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 10]
```diff
@@ -4602,23 +4602,23 @@
...
+  long_name = ice nucleation number in MG MP
+  units = kg-1
   dimensions = (horizontal_dimension,vertical_dimension)
   type = real
   kind = kind_phys
 [ccn_nm]
-  standard_name = ccn_number_concentration
-  long_name = CCN number concentration
-  units = kg-1?
+  standard_name = tendency_of_ccn_activated_number
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta:4612]: Same as for ccpp-physics, has this change been coordinated with @grantfirl and @ligiabernardet?

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 11]
```diff
@@ -6579,54 +6579,6 @@
 [ccpp-arg-table]
   name = GFS_interstitial_type
   type = ddt
-[qv_r]
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert the changes between lines 6582 and 6629.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 12]
```diff
@@ -6651,14 +6603,6 @@
   type = real
   kind = kind_phys
   active = (flag_for_microphysics_scheme == flag_for_fer_hires_microphysics_scheme)
-[cwm]
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 13]
```diff
@@ -7619,7 +7563,7 @@
   dimensions = (horizontal_dimension)
   type = real
   kind = kind_phys
-[fscav]
+[fscav_cs]
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 14]
```diff
@@ -9218,15 +9162,15 @@
   units = none
   dimensions = ()
   type = integer
-[cldtausw]
+[pcldtausw]
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 15]
```diff
@@ -9218,15 +9162,15 @@
...
   standard_name = RRTMGP_cloud_optical_depth_layers_at_0_55mu_band
   long_name = approx .55mu band layer cloud optical depth
   units = none
   dimensions = (horizontal_dimension,vertical_dimension)
   type = real
   kind = kind_phys
   optional = F
-[cldtaulw]
+[pcldtaulw]
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Please revert this change.

**Review by climbfuji — CHANGES_REQUESTED** [review]
Same as for ccpp-physics, a number of changes need to be reverted, they don't belong into this PR and will cause conflicts later on when merging HWRF physics etc. Others are ok even though not within the scope of this PR.

**Comment by mzhangw** [comment]
I will do it today.

File: gfsphysics/GFS_layer/GFS_typedefs.meta (hunk 15 above)
Comment (mzhangw) [reply gfsphysics/GFS_layer/GFS_typedefs.meta]: cldtaulw is double defined.

File: gfsphysics/GFS_layer/GFS_typedefs.meta (hunk 14 above)
Comment (mzhangw) [reply gfsphysics/GFS_layer/GFS_typedefs.meta]: cldtausw is double defined

**Comment by mzhangw** [comment]
I reverted changes per request. It passed all RTs and is ready for review.

**Comment by climbfuji** [comment]
This PR was pulled into https://github.com/NCAR/fv3atm/pull/64, which merges the changes into NCAR's feature/transition-to-capgen-1 branch. These changes will come to develop in the next couple of weeks.

**Comment by climbfuji** [comment]
All changes from this PR were included in #186.

**Pull request closed without merging**